ProcessList.c

```diff
@@ -303,17 +304,48 @@ static void ProcessList_buildTree(ProcessList* this) {
       if (process->isRoot) {
          process = (Process*)Vector_get(this->processes, i);
          process->indent = 0;
-         process->tree_depth = 0;
          Vector_add(this->displayList, process);
          ProcessList_buildTreeBranch(this, process->pid, 0, 0, process->showChildren);
          continue;
       }
    }
 
+   // Under some ptrace(2) implementations, a process ptrace(2)-attaching a
+   // parent process in its process tree will causing the traced process to
+   // be re-parented to the tracing process; this will creating a loop in the
+   // process tree. In this case build separated tree(s) for loop(s) left
+   // here.
+   int display_size = Vector_size(this->displayList);
+   if(display_size < vsize) {
+      for (int i = 0; i < vsize; i++) {
+         Process* process = (Process*)Vector_get(this->processes, i);
+         if(!process->isRoot && process->tree_depth == 0) {
```

Comment:
```suggestion
         if (!process->isRoot && process->tree_depth == 0) {
```

Reply:
I think it may be worth considering to invert this condition to allow skipping the entire loop body that follows and thus reduce the overall indentation level.